addons/repo-proxy/common/src/main/java/org/commonjava/indy/repo/proxy/ContentBrowseRemoteIndyListingRewriteManager.java

```diff
@@ -129,37 +142,63 @@ public boolean rewriteResponse( HttpServletRequest request, HttpServletResponse
 
         logger.trace( "Start to get remote indy content list for indy {} and path {}", remoteIndyUrl, path );
 
-        URL remoteIndyURL = new URL( remoteIndyUrl );
-        String remoteIndyHost = remoteIndyURL.getHost();
-        int remotePort = remoteIndyURL.getPort();
-        // 80 or 443 can be ignored in http(s) url
-        if ( remotePort != 80 && remotePort != 443 && remotePort > 0 )
-        {
-            remoteIndyHost = remoteIndyHost + ":" + remotePort;
-        }
-        String localProxyHost = request.getServerName();
-        int localPort = request.getServerPort();
-        if ( localPort != 80 && localPort != 443 && localPort > 0 )
+        try
         {
-            localProxyHost = localProxyHost + ":" + localPort;
-        }
-        final StoreKey originalStoreKey = StoreKey.fromString( originalRepo.get() );
-        final String remoteContentResult = getRemoteIndyListingContent( originalStoreKey, path );
+            URL remoteIndyURL = new URL( remoteIndyUrl );
+            String remoteIndyHost = remoteIndyURL.getHost();
+            int remotePort = remoteIndyURL.getPort();
+            // 80 or 443 can be ignored in http(s) url
+            if ( remoteIndyUrl.contains( ":" ) && remotePort != 80 && remotePort != 443 && remotePort > 0 )
+            {
+                remoteIndyHost = remoteIndyHost + ":" + remotePort;
+            }
+            String localProxyHost = request.getServerName();
+            int localPort = request.getServerPort();
+            if ( localPort != 80 && localPort != 443 && localPort > 0 )
+            {
+                localProxyHost = localProxyHost + ":" + localPort;
+            }
+            final StoreKey originalStoreKey = StoreKey.fromString( originalRepo.get() );
+            final String remoteContentResult = getRemoteIndyListingContent( originalStoreKey, path );
 
-        if ( StringUtils.isNotBlank( remoteContentResult ) )
+            if ( StringUtils.isNotBlank( remoteContentResult ) )
+            {
+                logger.trace( "Start replacing original host {} to target host {}", remoteIndyHost, localProxyHost );
+                final String replaced =
+                        RepoProxyUtils.replaceAllWithNoRegex( remoteContentResult, remoteIndyHost, localProxyHost );
+                logger.trace( "Replaced result for remote indy content list: {}", replaced );
+                switch ( httpMethod )
+                {
+                    case HttpMethod.GET:
+                        logger.trace( "Handle get request for content browse rewrite" );
+                        response.getWriter().write( replaced );
+                        return true;
+                    case HttpMethod.HEAD:
+                        logger.trace( "Handle head request for content browse rewrite" );
+                        Map<String, String> headers = getRemoteIndyListingHeaders( originalStoreKey, path );
```

Comment:
Wait...this seems to suggest that we're making TWO http calls to Indy for HEAD requests. This would indicate that we're doing a heavier process for head requests than get requests...which is backward.

If we can, I'd prefer to only do this listing-headers call for HEAD requests, even if that meant that we had to leave off the content-length header.

Reply:
Ok. I'll use just head request for HEAD case.